pkg/asset/installconfig/aws/validation.go

```diff
@@ -49,12 +49,32 @@ func Validate(ctx context.Context, meta *Metadata, config *types.InstallConfig)
 	allErrs = append(allErrs, validatePlatform(ctx, meta, field.NewPath("platform", "aws"), config.Platform.AWS, config.Networking, config.Publish)...)
 
 	if config.ControlPlane != nil && config.ControlPlane.Platform.AWS != nil {
-		allErrs = append(allErrs, validateMachinePool(ctx, meta, field.NewPath("controlPlane", "platform", "aws"), config.Platform.AWS, config.ControlPlane.Platform.AWS, controlPlaneReq)...)
+		allErrs = append(allErrs, validateMachinePool(ctx, meta, field.NewPath("controlPlane", "platform", "aws"), config.Platform.AWS, config.ControlPlane.Platform.AWS, controlPlaneReq, "")...)
 	}
+
 	for idx, compute := range config.Compute {
 		fldPath := field.NewPath("compute").Index(idx)
+
 		if compute.Platform.AWS != nil {
-			allErrs = append(allErrs, validateMachinePool(ctx, meta, fldPath.Child("platform", "aws"), config.Platform.AWS, compute.Platform.AWS, computeReq)...)
+			switch compute.Name {
+			// Pool's specific validation
+			// Edge Compute Pool: AWS Local Zones is valid only when installing in existing VPC.
+			case types.MachinePoolEdgeRoleName:
+				if len(config.Platform.AWS.Subnets) <= 0 {
+					// TODO create a test for this error
+					return errors.New(field.Required(fldPath, "invalid install config. edge machine pool is valid when installing in existing VPC.").Error())
+				}
+				edgeSubnets, err := meta.EdgeSubnets(ctx)
+				if err != nil {
+					allErrs = append(allErrs, field.Invalid(fldPath, edgeSubnets, err.Error()))
```

Comment:
Is it possible for this error message to be skipped when the len(edgeSubnets) <= 0?

Reply:
Yes, it seems. I wonder if it will be better to add the `return allErrs.ToAggregate()` here. wdyt?

Reply:
Changed [here](https://github.com/openshift/installer/pull/6371/commits/ceedde03d5151c7b68457a2e3ef7304553770141#diff-bab1b8072f369df53e27d65607145fb93caabd3a86fdefbcafb176e823726069R69). Let me know if it's what you were thinking.